server/api/crud/alerts.py

```diff
@@ -266,5 +262,23 @@ def reset_alert(self, session: sqlalchemy.orm.Session, project: str, name: str):
     def _delete_notifications(self, alert: mlrun.common.schemas.AlertConfig):
         for notification in alert.notifications:
             server.api.api.utils.delete_notification_params_secret(
-                alert.project, notification
+                alert.project, notification.notification
+            )
+
+    def _validate_and_mask_notificatsions(self, alert_data):
+        notifications = [
+            mlrun.common.schemas.notification.Notification(**notification.to_dict())
```

Comment:
Looking at this PR, I'm starting to wonder if it wasn't better to use the client-side `Notification` object in your structures. You're very busy transforming between those objects all the time going back-and-forth their dict representation.
It's worth considering, maybe will save a lot of this dirty work.